src/RA_CodeNotes.cpp

```diff
@@ -20,35 +22,39 @@ size_t CodeNotes::Load(const std::string& sFile)
     Clear();
 
     SetCurrentDirectory(NativeStr(g_sHomeDir).c_str());
-    FILE* pf = nullptr;
-    if (fopen_s(&pf, sFile.c_str(), "rb") == 0)
+
+    std::ifstream ifile{ sFile, std::ios::binary };
+
+    Document doc;
+    IStreamWrapper isw{ ifile };
+    doc.ParseStream(isw);
+
+    if (!doc.HasParseError())
+
     {
-        Document doc;
-        doc.ParseStream(FileStream(pf));
-        if (!doc.HasParseError())
-        {
-            ASSERT(doc["CodeNotes"].IsArray());
+        ASSERT(doc["CodeNotes"].IsArray());
 
-            const Value& NoteArray = doc["CodeNotes"];
+        // use auto when you can to prevent implicit conversions (you get warnings on higher levels)
+        const auto& NoteArray = doc["CodeNotes"];
 
-            for (SizeType i = 0; i < NoteArray.Size(); ++i)
-            {
-                const Value& NextNote = NoteArray[i];
-                if (NextNote["Note"].IsNull())
-                    continue;
+        for (auto& i : NoteArray.GetArray())
+        {
+            if (i["Note"].IsNull())
+                continue;
 
-                const std::string& sNote = NextNote["Note"].GetString();
-                if (sNote.length() < 2)
-                    continue;
+            const std::string& sNote = i["Note"].GetString();
+            if (sNote.length() < 2)
+                continue;
 
-                const std::string& sAddr = NextNote["Address"].GetString();
-                ByteAddress nAddr = static_cast<ByteAddress>(std::strtoul(sAddr.c_str(), nullptr, 16));
-                const std::string& sAuthor = NextNote["User"].GetString();	//	Author?
+            const std::string& sAddr = i["Address"].GetString();
+            auto nAddr = static_cast<ByteAddress>(std::strtoul(sAddr.c_str(), nullptr, 16));
 
-                m_CodeNotes.insert(std::map<ByteAddress, CodeNoteObj>::value_type(nAddr, CodeNoteObj(sAuthor, sNote)));
-            }
+            // Author?
+            const std::string& sAuthor = i["User"].GetString();
+
+            auto code_pair{ std::make_pair(nAddr, CodeNoteObj{ sAuthor, sNote }) };
```

Comment:
Your CodeNoteObj would need a move constructor and assignment for this to work properly. Instead of making the "ID" const just delete copying, that way you enforce uniqueness making duplication impossible. If you tried to define moving explicitly the program would not compile. Any way if you define and/or delete any default operation you need to define and/or delete them all.